app/models/annual_leave_request.rb

```diff
@@ -6,6 +6,7 @@ class AnnualLeaveRequest < ApplicationRecord
   validates :date_from, :date_to, comparison: { greater_than: Time.zone.today, message: "must be in the future" }
   validate :user_has_enough_annual_leave_remaining
   validates :confirm_approval, acceptance: { accept: "confirmed" }
+  validates :denial_reason, presence: true, allow_nil: true
```

Comment:
This allows me to create `AnnualLeaveRequest`s with no denial reason (default is `nil` on creation), whilst simultaneously checking a Line Manager gives a reason on denial. Is there a better way to do this?

Reply:
Ah that makes sense. That suggests you've got a problem with your database table design to me, because having to have a denial reason for requests that aren't denied is a sign the modelling isn't perfect. I've done things like move the state updates into a different table or even have a separate table for records for each state in other systems before, but it's hard to say exactly what the best choice would be here.

Modelling state machines in databases is tricky though and you might want to leave it as is for now until it starts to cause more problems.

Reply:
I'm going to merge this request and make a trello ticket to revisit this topic